### PR TITLE
feat: check contract on all EVM chains

### PR DESCRIPTION
**Pull request by HenryQW**
also refactored protocol.network

### CODE REVIEW

File: common/worker/name_service/address.go [hunk 1]
```diff
@@ -3,31 +3,47 @@ package name_service
...
+func CheckContractOnEVM(address string) []string {
+	var wg sync.WaitGroup
+	var result []string
+	for _, n := range protocol.EVMNetworks {
+		wg.Add(1)
+		network := n
+		go func() {
+			defer wg.Done()
+			if !IsContract(address, network) {
+				result = append(result, network)
```
Comment (kallydev) [inline common/worker/name_service/address.go]: append is non-thread safe.

Comment (kallydev) [reply common/worker/name_service/address.go]: replace append with index to fix this problem.

```go
// Map manipulates a slice and transforms it to a slice of another type.
func Map[T any, R any](collection []T, iteratee func(T, int) R) []R {
	result := make([]R, len(collection))

	for i, item := range collection {
		result[i] = iteratee(item, i)
	}

	return result
}
```

File: service/hub/internal/server/middlewarex/apiHandleMiddleware.go [hunk 2]
```diff
@@ -57,6 +69,23 @@ func CheckAPIKeyMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
+func CheckContractMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (kallydev) [inline service/hub/internal/server/middlewarex/apiHandleMiddleware.go]: does this middleware support batch notes request?

**Review by kallydev — CHANGES_REQUESTED** [review]

File: common/worker/name_service/address.go (hunk 1 above)
Comment (HenryQW) [reply common/worker/name_service/address.go]: how about a simple `sync.Mutex` lock?

File: service/hub/internal/server/middlewarex/apiHandleMiddleware.go (hunk 2 above)
Comment (HenryQW) [reply service/hub/internal/server/middlewarex/apiHandleMiddleware.go]: ~no~ supported now

**Pull request closed without merging**